lib/Path/Dispatcher/Dispatch.pm

```diff
@@ -1,12 +1,13 @@
 package Path::Dispatcher::Dispatch;
-use Any::Moose;
+use Carp;
+use Moo;
 use Try::Tiny;
 
 use Path::Dispatcher::Match;
 
 has _matches => (
     is        => 'ro',
-    isa       => 'ArrayRef',
+    isa       => sub { die "not an ArrayRef" unless 'ARRAY' eq ref $_[0] },
```

Comment:
@sartak is this the kind of  type constraint change you are concerned about ?
Where I replace a named type with a coderef to check the type.

The reason is that Moo does not have a type system, so you can't just name a type. My understanding is that usual practice is to provide a coderef instead, but I would be open to an alternative suggestion. I am familiar with Moose, but this is the first time I have worked with Moo, so there could be better practice out there that I should be using, eg Type::Tiny